transaction/core/src/input_rules.rs

```diff
@@ -68,6 +70,21 @@ pub struct InputRules {
 }
 
 impl InputRules {
+    /// Get all tx target keys associated to outputs appearing in these rules
+    pub fn associated_tx_target_keys(&self) -> BTreeSet<CompressedRistrettoPublic> {
```

Comment:
This should probably just get the set of all outputs appearing in the rules instead